deploy/role.yaml

```diff
@@ -105,12 +105,18 @@ rules:
   - statefulsets
   verbs:
   - '*'
+- apiGroups:
```

Comment:
I broke this out into it's own section and limited access to the 2 current `cert-manager.io` resources we need access to, which is `Certificates` and `Issuers`. We'll need to extend this if we ever do more elaborate certificate management.